Sources/StreamChat/Database/DTOs/MessageDTO.swift

```diff
@@ -531,7 +531,12 @@ class MessageDTO: NSManagedObject {
 extension MessageDTO {
     /// A possible additional local state of the message. Applies only for the messages of the current user.
     var localMessageState: LocalMessageState? {
-        get { localMessageStateRaw.flatMap(LocalMessageState.init(rawValue:)) }
+        get {
+            if attachments.filter({ $0.localState == .uploadingFailed }).isEmpty == false {
```

Comment:
Iterating through the attachments on each access to the local message state - could this have performance implications?